src/main/java/com/andrzejewski/todolist/task/TaskService.java

```diff
@@ -31,9 +34,15 @@ public boolean deleteTask(Long userId, Long taskId) {
     }
 
     @Transactional
-    public void updateTask(Long id, String text, LocalDateTime date) {
+    public void updateTaskText(Long id, String text) {
+        TaskEntity taskEntity = mTaskRepository.findById(id).orElseThrow(() -> new TaskDoesNotExistException(id));
+        taskEntity.setTaskText(text);
+    }
+
+    @Transactional
+    public void markTaskAsDone(Long id, String taskCompletionDate) {
         TaskEntity taskEntity = mTaskRepository.findById(id).orElseThrow(() -> new TaskDoesNotExistException(id));
-        if (text != null) { taskEntity.setTaskText(text); }
-        taskEntity.setTaskCompletionDate(date);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-ddHH:mm:ss");
```

Comment:
This should be done in controller. You can just use `LocalDateTime` as controller parameter and Spring will handle the conversion.